src/mgr/ActivePyModules.cc

```diff
@@ -511,12 +503,8 @@ PyObject *ActivePyModules::get_python(const std::string &what)
     });
   } else {
     derr << "Python module requested unknown data '" << what << "'" << dendl;
-    without_gil_t no_gil;
-    with_gil_t with_gil{no_gil};
     Py_RETURN_NONE;
   }
-  without_gil_t no_gil;
-  no_gil.acquire_gil();
   if(ttl_seconds) {
     return jf.get();
   } else {
```

Comment:
Is this unnecessary? PyObject::get calls Py_INCREF. My understanding is that any operation on PyObjects should acquire the gil.

Reply:
It is true that any operation on PyObjects requires the gil, it's just that we necessarily already have the gil here, so this isn't doing anything useful (these two lines effectively release the gil and then reacquire it). Notice the definition of the `without_gil_t` constructor:

```python
without_gil_t::without_gil_t()
{
  assert(PyGILState_Check());
  release_gil();
}
```
The assertion fails when the calling thread doesn't have the gill, so an instance of `without_gil_t` can't be created without the gil. Any back-to-back invocations of:

```python
without_gil_t no_gil;
no_gil.acquire_gil();
```
are always unnecessary. In cases where we just want to ensure that we didn't make any mistakes in our evaluation of whether the code will have the gil at a particular point, I suggest we just use the assertion:

```python
assert(PyGILState_Check());
```

Reply:
You're right. Thanks for the explanation.